File: BlueTeam/dashboard/src/types/validation.ts
export interface ValidationChecks {
  detection: boolean;
  risk: boolean;
  alert: boolean;
  incident: boolean;
  response: boolean;
}

export interface ValidationRecord {
  validation_id: string;
  event_id: string;
  attack_type: string;
  checks: ValidationChecks;
  overall_result: 'PASS' | 'FAIL';
  reasons: string[];
  created_at: string;
}

export interface ReplayTimelineEvent {
  stage: string;
  timestamp: string;
  status: string;
  detail?: string;
  rule?: string;
  score?: number;
  mode?: string;
  validation_id?: string;
}

export interface ReplayResponse {
  validation_id: string | null;
  event_id: string;
  timeline: ReplayTimelineEvent[];
}

File: BlueTeam/api/schemas/validation.py
from typing import Any, Dict, List, Optional
from datetime import datetime
from pydantic import BaseModel


class ExpectedConfigSchema(BaseModel):
    attack_type: str
    detection: Optional[str] = None
    risk_level: Optional[str] = None
    alert: Optional[bool] = None
    incident: Optional[bool] = None
    response: Optional[str] = None


class ValidationRequestSchema(BaseModel):
    event_id: str
    expected: Optional[ExpectedConfigSchema] = None


class ValidationResponseSchema(BaseModel):
    validation_id: str
    event_id: str
    attack_type: str
    checks: Dict[str, bool]
    overall_result: str
    reasons: List[str]
    created_at: datetime


class ReplayTimelineEvent(BaseModel):
    stage: str
    timestamp: str
    status: str
    detail: Optional[str] = None
    rule: Optional[str] = None
    score: Optional[int] = None
    mode: Optional[str] = None
    validation_id: Optional[str] = None


class ReplayResponseSchema(BaseModel):
    validation_id: Optional[str] = None
    event_id: str
    timeline: List[ReplayTimelineEvent]

